src/autodiff.jl

```diff
@@ -89,22 +87,32 @@ Creates an object for computing Hessian adjoint tangent projections.
 * `varx::SubT`: View of `map` on `x`
 * `t1svarx::SubD`: Active (AD) view of `map` on `x`
 """
-struct Hessian{Func, VI, VT, VHP, VP, VD, SubT, SubD} <: AbstractHessian
+struct Hessian{Func, VI, VT, VHP, VP, VD, SubT, SubD, Buff} <: AbstractHessian
     func::Func
     host_t1sseeds::VHP # Needed because seeds have to be created on the host
     t1sseeds::VP
+    x::VT
     t1sF::VD
     ∂t1sF::VD
-    x::VT
     t1sx::VD
     ∂t1sx::VD
     map::VI
     varx::SubT
     t1svarx::SubD
+    buffer::Buff
 end
 
+# Cache for adjoint
+# Largely inspired from ChainRulesCore.jl:
+# https://juliadiff.org/ChainRulesCore.jl/stable/design/changing_the_primal.html#The-Journey-to-rrule
+struct PullbackMemory{F, S, I}
```

Comment:
This makes total sense now. You essentially came up with the stack that is needed to save the entire state of the computation. That is: function, value, partial. So what is called stack here should probably be a partial. And `PullbackMemory` is what is called tape or stack in AD.